code/modules/mob/mob_helpers.dm

```diff
@@ -361,9 +361,14 @@ var/list/intents = list(I_HELP,I_DISARM,I_GRAB,I_HARM)
 	set name = "Rest"
 	set category = "IC"
 
-	resting = !resting
-	update_canmove()
-	to_chat(src, "<span class='notice'>You are now [resting ? "resting" : "getting up"].</span>")
+	if(!resting && (world.time > rest_CD))
+		rest_CD = world.time + 20
+		to_chat(src, "<span class='notice'>You are now resting.</span>")
+		StartResting()

```

Comment:
Why are these separate CD's even necessary? I was under the direct impression that the reason for setting the move delay under `update_canmove` was to avoid having this altogether.